tests/unit/acquisition/function/test_entropy.py

```diff
@@ -519,3 +525,203 @@ def test_batch_gibbon_is_sum_of_individual_gibbons_and_repulsion_term(
+        ThompsonSamplerFromTrajectory(sample_min_value=False),
+    ],
+)
+def test_mumbo_raises_if_passed_sampler_with_sample_min_value_False(
+    sampler: ThompsonSampler[MUMBOModelType],
+) -> None:
+    search_space = Box([0, 0], [1, 1])
+    with pytest.raises(ValueError):
+        MUMBO(search_space, min_value_sampler=sampler)
+
+
+def test_mumbo_default_sampler_is_exact_thompson() -> None:
+    search_space = Box([0, 0], [1, 1])
+    builder = MUMBO(search_space)
+    assert isinstance(builder._min_value_sampler, ExactThompsonSampler)
+    assert builder._min_value_sampler._sample_min_value
+
+
+@pytest.mark.parametrize(
+    "sampler",
+    [
+        ExactThompsonSampler(sample_min_value=True),
+        GumbelSampler(sample_min_value=True),
+        ThompsonSamplerFromTrajectory(sample_min_value=True),
+    ],
+)
+def test_mumbo_initialized_with_passed_sampler(sampler: ThompsonSampler[MUMBOModelType]) -> None:
+    search_space = Box([0, 0], [1, 1])
+    builder = MUMBO(search_space, min_value_sampler=sampler)
+    assert builder._min_value_sampler == sampler
+
+
+def test_mumbo_raises_when_use_trajectory_sampler_and_model_without_trajectories() -> None:
+    dataset = Dataset(tf.zeros([3, 2], dtype=tf.float64), tf.ones([3, 2], dtype=tf.float64))
+    search_space = Box([0, 0], [1, 1])
+    builder = MUMBO(  # type: ignore
+        search_space, min_value_sampler=ThompsonSamplerFromTrajectory(sample_min_value=True)
+    )
+    model = MultiFidelityQuadraticMeanAndRBFKernel()
+    with pytest.raises(ValueError):
+        builder.prepare_acquisition_function(model, dataset=dataset)  # type: ignore
+
+
+@unittest.mock.patch("trieste.acquisition.function.entropy.mumbo")
+@pytest.mark.parametrize(
+    "min_value_sampler",
+    [ExactThompsonSampler(sample_min_value=True), GumbelSampler(sample_min_value=True)],
+)
+def test_mumbo_builder_builds_min_value_samples(
+    mocked_mves: MagicMock,
+    min_value_sampler: ThompsonSampler[SupportsCovarianceWithTopFidelity],
+) -> None:
+    dataset = Dataset(tf.zeros([3, 2], dtype=tf.float64), tf.ones([3, 2], dtype=tf.float64))
+    search_space = Box([0, 0], [1, 1])
+    builder = MUMBO(search_space, min_value_sampler=min_value_sampler)
+    model = MultiFidelityQuadraticMeanAndRBFKernelWithSamplers(dataset)
+    model.kernel = (
+        gpflow.kernels.RBF()
+    )  # need a gpflow kernel object for random feature decompositions
+    builder.prepare_acquisition_function(model, dataset=dataset)
+    mocked_mves.assert_called_once()
+
+    # check that the Gumbel samples look sensible
+    min_value_samples = mocked_mves.call_args[0][1]
+    query_points = builder._search_space.sample(num_samples=builder._grid_size)
+    query_points = tf.concat([dataset.query_points, query_points], 0)
+    query_points = add_fidelity_column(query_points[:, :-1], model.num_fidelities - 1)
+    fmean, _ = model.predict(query_points)
+    assert max(min_value_samples) < min(fmean)
```

Comment:
I don't see a theoretical reason why this should always be true... it is a bit of a weird thing to test but I guess it does the job of checking that `min_value_samples` returns small values?

A better test would have been to do `model.sample(query_points)` and check if the empirical distribution of the minimum of the samples resembles `min_value_samples`.